modules/standard/CommDiagnostics.chpl

```diff
@@ -372,6 +372,54 @@ module CommDiagnostics
   }
 
 
+  /*
+    Print the current communication counts in a text table showing
+    locales x operations.  By default, operations with a count of
+    zero will not be displayed in the table.
+
+    :arg allOps: Print all operations regardless of count? (defaults to ``false``)
+    :type allOps: `bool`
+  */
+  proc printCommDiagnosticsTable(allOps=false) {
+    use Reflection;
+
+    // grab all comm diagnostics
+    var CommDiags = getCommDiagnostics();
+
+    // cache number of fields and store vector of whether field is active
+    param nFields = numFields(chpl_commDiagnostics);
+    var fieldInUse: [0..<nFields] bool;
+
+    // print column headers while determining which fields are active
+    writef("%7s", "locale");
+    for param fieldID in 0..<nFields {
+      param name = getFieldName(chpl_commDiagnostics, fieldID);
+      var found = false;
+      for locID in LocaleSpace {
+        const val = getField(CommDiags[locID], fieldID);
+        if allOps || val != 0 {
+          if (found == false) {
+            found = true;
+            fieldInUse[fieldID] = true;
+            writef("%16s", name);
+          }
+        }
+      }
+    }
+    writeln();
+
+    // print a row per locale showing the active fields
+    for locID in LocaleSpace {
+      writef("%7s", locID:string);
+      for fieldID in 0..<nFields {
+        if fieldInUse[fieldID] {
+          writef("%16s", getField[CommDiags[locID], fieldID):string);
```

Comment:
`getField[` -> `getField(` and I think fieldID needs to be `param`. Nit, but seems like you could just use %16i and not do the string cast